app/scripts/metamask-controller.js

```diff
@@ -242,11 +241,17 @@ export default class MetamaskController extends EventEmitter {
           };
           return cb(modifiedNetworkState);
         }),
-      onPreferencesStateChange: this.preferencesController.store.subscribe.bind(
-        this.preferencesController.store,
-      ),
+      onPreferencesStateChange: (cb) =>
+        this.preferencesController.store.subscribe((preferencesState) => {
+          const modifiedPreferencesState = {
+            ...preferencesState,
+            useStaticTokenList: !this.preferencesController.store.getState()
```

Comment:
I won't block on this at some point I think we should just switch the TokenListController to expect the `useTokenDection` value instead.

Reply:
The changes for the switch were dropped as it will be a breaking change in controllers repo for both mobile and extension at that time. [#560](https://github.com/MetaMask/controllers/pull/560) tracks the related changes if we decided to move forward with that.